influxql/iterator.go

```diff
@@ -259,9 +261,11 @@ func newAuxIteratorFields(seriesKeys SeriesList, opt IteratorOptions) auxIterato
 
 func (a auxIteratorFields) close() {
 	for _, f := range a {
+		f.mtx.Lock()
 		for _, itr := range f.itrs {
 			itr.Close()
 		}
+		f.mtx.Unlock()
```

Comment:
This is `auxIteratorFields` which doesn't have a `Close()` method. We could make a close method and put the entire for loop in there, but I'm not sure that's a meaningful difference.